content/cloud-docs/api-docs/providers.mdx

```diff
@@ -336,6 +726,559 @@ curl \
+
+Creates a new registry provider platform. This endpoint only applies to private providers.
+
+| Status  | Response                                                      | Reason                                                         |
+| ------- | ------------------------------------------------------------- | -------------------------------------------------------------- |
+| [201][] | [JSON API document][] (`type: "registry-provider-platforms"`) | Successfully published provider platform                       |
+| [422][] | [JSON API error object][]                                     | Malformed request body (missing attributes, wrong types, etc.) |
+| [403][] | [JSON API error object][]                                     | Forbidden - not available for public providers                 |
+| [404][] | [JSON API error object][]                                     | User not authorized                                            |
+
+### Request Body
+
+This POST endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path                   | Type   | Default  | Description                              |
+| -------------------------  | ------ | -------  | -------------------------------------    |
+| `data.type`                | string |          | Must be `"registry-provider-platforms"`. |
+| `data.attributes.os`       | string |          | A valid semver version string.           |
+| `data.attributes.arch`     | string |          | A valid os string.                       |
+| `data.attributes.shasum`   | string |          | A valid shasum string.                   |
+| `data.attributes.filename` | string |          | A valid filename string.                 |
+
+
+### Sample Payload
+
+```json
+{
+  "data": {
+    "type": "registry-provider-version-platforms",
+    "attributes": {
+      "os": "linux",
+      "arch": "amd64",
+      "shasum": "8f69533bc8afc227b40d15116358f91505bb638ce5919712fbb38a2dec1bba38",
+      "filename": "terraform-provider-aws_3.1.1_linux_amd64.zip"
+    }
+  }
+}
+```
+
+### Sample Request
+
+```shell
+curl \
+  --header "Authorization: Bearer $TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @payload.json \
+  https://app.terraform.io/api/v2/organizations/hashicorp/registry-providers/private/hashicorp/aws/versions/3.1.1/platforms
+```
+
+### Sample Response
+
+```json
+{
+  "data": {
+    "id": "provpltfrm-BLJWvWyJ2QMs525k",
+    "type": "registry-provider-platforms",
+    "attributes": {
+      "os": "linux",
+      "arch": "amd64",
+      "filename": "terraform-provider-aws_3.1.1_linux_amd64.zip",
+      "shasum": "8f69533bc8afc227b40d15116358f91505bb638ce5919712fbb38a2dec1bba38",
+      "permissions": {
+        "can-delete": true,
+        "can-upload-asset": true
+      },
+      "provider-binary-uploaded": false
```

Comment:
Would the response be "provider-binary-uploaded": true ? Or does the curl on 796 not upload the binary but just adds the name etc? I know. we just did this, but since I did the curl differently... I am not sure.
If the curl above does upload the binary, it could also be helpful to change line 831 to be "provider-binary-download"

Reply:
The [Upload provider binary curl request](https://gist.github.com/jbonhag/362cef97e1389a5d2bc7cab87fa1ba79#upload-provider-binary) happens after this step